### PR TITLE
VB-1434 Add code to server side to allow for bi-weekly session templates

### PR DESCRIPTION
**Pull request by AledWEvans**
## What does this pull request do?

Adds bi-weekly functionality

## What is the intent behind these changes?

allow configurations for bristol

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/PrisonerSessionValidatorTest.kt [hunk 1]
```diff
@@ -30,22 +30,35 @@ class PrisonerSessionValidatorTest {
     private val prisonerDetail3LevelPrison = PrisonerDetailDto("A4000001", unitCode1 = "C", unitCode2 = "1", unitCode3 = "004", unitCode4 = null)
     @Test
     fun `session available to all prisoners when permitted session location is empty`() {
-      val sessionTemplate = createSessionTemplate(permittedSessionLocations = mutableListOf())
+      // Given
```
Comment (AledWEvans) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/PrisonerSessionValidatorTest.kt:33]: Sorry @dhirajshettymoj I have update ur unit test always like the  //Given//Then//When... it's a standard approach in this code base.. I like it, makes code readable.

Comment (dhirajshettymoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/PrisonerSessionValidatorTest.kt:33]: Thanks! Yes, it's more readable.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/SessionDatesUtil.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+    return biWeeklyFirstBookableSessionDay.datesUntil(
+      adjustedLastBookableSessionDay, Period.ofWeeks(2)
+    )
+  }
+
+  private fun isSkipWeek(
+    validFromDate: LocalDate,
+    firstBookableSessionDay: LocalDate
+  ) = WEEKS.between(validFromDate, firstBookableSessionDay).toInt() % 2 != 0
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/SessionDatesUtil.kt:48]: the 2nd parameter to between is exclusive so should it be firstBookableSessionDay.plusDays(1)?

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**